toolbox/py_xnvme_pkg/xnvme/library_loader.py

```diff
@@ -0,0 +1,50 @@
+"""
+    The xnvme.library_loader provides a single method to dynamically load the xNVMe
+    shared library. This intent is that the loader is invoked by the
+    'xnvme.ctypes_mapping' on package-initialization, such that the user can directly
+    get a library-handle like so:
+
+        from xnvme.ctypes_mapping import capi
+
+    This is useful since usage of the library handle can have multiple entry points.
+"""
+import ctypes
+import ctypes.util
+import os
+import subprocess
+
+
+def search_paths():
+    """Generate paths to try and load"""
+
+    for search in ["xnvme-shared", "xnvme_shared"]:
+        path = ctypes.util.find_library(search)
+        if path:
+            yield path
+
+    try:
+        proc = subprocess.run(
+            ["pkg-config", "xnvme", "--variable=libdir"],
+            check=True,
+            capture_output=True,
+        )
+        if not proc.returncode:
+            yield os.path.join(
+                proc.stdout.decode("utf-8").strip(), "libxnvme-shared.so"
+            )
+    except subprocess.CalledProcessError:
+        pass
+
```

Comment:
I would recommend adding a third-priority path with the local build: `yield f"{git_root}/builddir/lib/libxnvme-shared.dylib"`

Reply:
I don't like having a dependency on the source-repository in the library loader. Libraries should be installed such that ``pkg-config`` and regular tooling can locate them.
What I have done instead is add a platform specific extension, such that it loads using ``.so``, ``.dll`` and ``.dylib`` depending the platform it is running on.